### ISSUE TITLE
Stop using cookies for PEPhub authentication

### ISSUE THREAD

**Issue by nleroy917** [pepkit/pephub#96]
See my proposal, background, and rationale here: #95

**Comment by nleroy917** [comment]
@nsheff @Khoroshevskyi 
**I am still thinking about how other clients will manage authentication with pephub.**

If a new client (say a `cli`) authenticates with pephub, currently, we [redirect to the **`/{namespace}`** page of the logged in user](https://github.com/pepkit/pephub/blob/571ffed256dbfac63b2f6d772948be7cb5066f93/pephub/routers/auth/base.py#L87). When doing this, we only update the cookie and send it back. This works well for our architecture where we have the UI served with the server, so we automatically get it back on the browser with zero work. However, for decoupled clients (i.e. a new server or `cli`), they have no way of extracting out the JWT from the cookies. Further, redirecting to the user `/{namespace}` page doesn't make sense for headless clients or clients that don't want to use the user interface.

I think a better way is to do one of two things: 
1. Send back to a generic, blank HTML page with the JWT in the URL params
2. Allow the authenticating client to specify a "send-to" link that sends them back where they originally came from, again with the JWT in the URL params

Something like:
`http://localhost:1234?access_token=ey.754821fhjd....`

**Simply put, I don't think it makes sense to *always* send clients (`cli`, user in a browser) back to the `/{namespace}` page. This authentication flow also doesn't make it easy to get the JWT**

**Comment by nleroy917** [comment]
based on some conversations, we should do one of two things:
1. Authenticate via popup and respond with a JSON blob that has the token, or
2. Look for a `client_redirect` parameter and send back there after authentication is complete.

**Comment by nleroy917** [comment]
I went with option 2. On the pephub UI, I link to the login endpoint via the following:
```html
<a href="/auth/login?client_redirect_uri=/login/success">
 Login
</a>
```
Notice the `client_redirect_uri=/login/success` query param. This tells the server to send me back there after all the authentication is complete. I am serving a custom HTML page there that will look for `token=ey742y1uhfr2....` in the URL, extract it, store it, and then send you somewhere else.

With this new method, the server is agnostic about how the client handles the token once it is received. It no longer stores cookies. Instead, it gives it to you in the URL and it is now up to the client to do something with it.

I also provide a very simple HTML landing page where the token is provided if no `client_redirect_uri` is supplied:

<img width="1728" alt="image" src="https://user-images.githubusercontent.com/41063083/222837569-52dda864-792c-4e05-bb02-4884e20f1d61.png">


Next steps are to not validate session info via cookies, but through authorization headers. Namely,

```json
{
  "Authorization": "Bearer ey78392yurh19...."
}
```

**Comment by nleroy917** [comment]
Switching over to `Authorization` headers threw a big wrench into the way we render and authorize the frontend. Since so much of the interface is rendered server-side and requires a cookie, I decided to create a new dependencies file that runs authorization flows using that cookie, instead of the header. **This is just for the UI**. The API is now completely using the `Authorization: Bearer` method with JWT's.

In summary, there are now two dependency files:
1. `dependencies.py`: This file is for the API and should be the default dependency injection functions
2. `view_dependencies.py`: This file is made specifically for the UI and injects dependencies based on the `pephub_session` cookie sent. It was made since much of the HTML on initial page load was written with server-side rendering in mind. All calls to the API to make data updates and some data fetches are using the API and for this we grab the cookie JWT and send it as a header.

**Comment by nleroy917** [comment]
While I have tested a lot of the features, we should all try it out to make sure it's functioning properly.

**Comment by nleroy917** [comment]
This was fixed in the latest release
